### PR TITLE
tls: make error message more concise

### PR DESCRIPTION
**Pull request by tyxia**
Fix: #29062

### CODE REVIEW

**Comment by tyxia** [comment]
/assign @ggreenway 

PTAL, Thanks

File: source/extensions/transport_sockets/tls/ssl_socket.cc [hunk 1]
```diff
@@ -233,11 +233,7 @@ void SslSocket::drainErrorQueue(bool syscall_error_occurred) {
     if (failure_reason_.empty()) {
       failure_reason_ = "TLS error:";
     }
-    failure_reason_.append(
-        "SSL_ERROR_SYSCALL error has occured, which indicates the operation failed externally to "
-        "the library. This is typically |errno| but may be something custom if using a custom "
-        "|BIO|. It may also be signaled if the transport returned EOF, in which case the "
-        "operation's return value will be zero.");
+    failure_reason_.append("SSL_ERROR_SYSCALL, check openssl doc for more details");
```
Comment (ggreenway) [inline source/extensions/transport_sockets/tls/ssl_socket.cc:236]:
```suggestion
    failure_reason_.append("SSL_ERROR_SYSCALL");
```

**Review by ggreenway — CHANGES_REQUESTED** [review]
Asking end users to refer to openssl docs in the access logs isn't helpful

/wait

**Review by ggreenway — APPROVED** [review]

**Pull request merged**